### PR TITLE
Solution to problem 5, already tested for 404 on line 25 of test suite

### PR DESCRIPTION
**Pull request by EdJackson98**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -71,4 +71,18 @@ describe("GET: /api/articles/:article_id", () => {
...
+})
+
+describe('/api/articles', () => {
+    test('GET:200 sends an array of article objects with the correct keys', () => {
+      return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then((response) => {
+          expect(response.body.articles).toEqual(expect.any(Array));
+          expect(Object.keys(response.body.articles[0])).toEqual(
```
Comment (leekli) [inline __tests__/app.test.js:84]: An extra assertion here to check how many objects are in the array (length) may be useful here?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -71,4 +71,18 @@ describe("GET: /api/articles/:article_id", () => {
...
+            expect.arrayContaining(['author', 'title', 'article_id', 'topic', 'created_at', 'votes', 'article_img_url', 'comment_count'])
+          );
+        });
+    });
 })
```
Comment (leekli) [inline __tests__/app.test.js:98]: The ticket 5 instructions state:

`In addition:

the articles should be sorted by date in descending order.
there should not be a body property present on any of the article objects.`

So there are 2 more tests to do:
* to check the articles are in created_at order
* to check the articles are in descending order

jest-sorted will help with this.

File: app.js [hunk 3]
```diff
@@ -14,6 +14,7 @@ const {
 app.get("/api/topics", getTopics);
 app.get('/api', getEndpoints);
 app.get("/api/articles/:article_id", getArticleByID)
+app.get("/api/articles", getAllArticles)
```
Comment (leekli) [inline app.js:17]: Good controller function name

File: controllers/articles.controller.js [hunk 4]
```diff
@@ -11,11 +11,10 @@ exports.getArticleByID = (req, res, next) => {
 
 };
 
-
-
 exports.getAllArticles = (req, res, next) => {
     fetchAllArticles()
     .then((articles) => {
+        console.log(articles)
```
Comment (leekli) [inline controllers/articles.controller.js]: Is this `console.log()` needed anymore?

File: models/articles.model.js [hunk 5]
```diff
@@ -13,4 +13,11 @@ exports.fetchArticleByID = (article_id) => {
             }
         return article
         });
+}
+
+exports.fetchAllArticles = () => {
+    let query = 'SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY articles.created_at DESC;'
```
Comment (leekli) [inline models/articles.model.js:19]: Brilliant SQL query - use of the `COUNT` aggregate function and `LEFT JOIN` will make this work a treat, well done!

**Review by leekli — CHANGES_REQUESTED** [review]
Hi Ed,

Well done so far! You're almost there.

* There's a console log to remove
* And there's 2 more tests you need to implement

See the individual comments on the request throughout.

Please implement the requested changes, and then submit another pull request for this ticket, for another review. Thanks.

**Review by Rayhaan2k9 — APPROVED** [review]
Great work fixing the suggestions in the feedback! Merge away! 😄

**Pull request merged**